src/Domain/MigrationStep/s150_ProductVariationMigration/VariantGroup/VariantGroupRepository.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\PimMigration\Domain\MigrationStep\s150_ProductVariationMigration\VariantGroup;
+
+use Akeneo\PimMigration\Domain\Command\Api\GetAttributeCommand;
+use Akeneo\PimMigration\Domain\Command\ChainedConsole;
+use Akeneo\PimMigration\Domain\Command\MySqlExecuteCommand;
+use Akeneo\PimMigration\Domain\Command\MySqlQueryCommand;
+use Akeneo\PimMigration\Domain\MigrationStep\s150_ProductVariationMigration\VariantGroup;
+use Akeneo\PimMigration\Domain\Pim\DestinationPim;
+
+/**
+ * Aims to read an write data related to the migration of the variant groups on the destination PIM.
+ *
+ * @author    Laurent Petard <laurent.petard@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ */
+class VariantGroupRepository
+{
+    /** @var ChainedConsole */
+    private $console;
+
+    /** @var int */
+    private $invalidVariantGroupTypeId;
+
+    public function __construct(ChainedConsole $console)
+    {
+        $this->console = $console;
+    }
+
+    public function retrieveNumberOfVariantGroups(DestinationPim $pim): int
+    {
+        return $this->retrieveNumberOfVariantGroupsByType($pim, 'VARIANT');
+    }
+
+    public function retrieveNumberOfRemovedInvalidVariantGroups(DestinationPim $pim): int
+    {
+        return $this->retrieveNumberOfVariantGroupsByType($pim, 'INVALID_VARIANT');
+    }
+
+    public function retrieveVariantGroups(DestinationPim $pim): \Traversable
+    {
+        $query =
+<<<SQL
+SELECT g.code,
+(SELECT COUNT(*) FROM pim_catalog_group_attribute axe WHERE axe.group_id = g.id) as nb_axes,
+(
+    SELECT COUNT(DISTINCT f.id)
+    FROM pim_catalog_group_product gp
+    INNER JOIN pim_catalog_product p ON gp.product_id = p.id
+    INNER JOIN pim_catalog_family f ON p.family_id = f.id
+    WHERE gp.group_id = g.id
+) as nb_families
+FROM pim_catalog_group_type gt
+INNER JOIN pim_catalog_group g ON g.type_id = gt.id
+WHERE gt.code = "VARIANT"
+SQL;
+
+        $variantGroups = $this->console->execute(new MySqlQueryCommand($query), $pim)->getOutput();
+
+        foreach ($variantGroups as $variantGroup) {
+            yield new VariantGroup($variantGroup['code'], (int) $variantGroup['nb_axes'], (int) $variantGroup['nb_families']);
+        }
+    }
+
+    public function retrieveVariantGroupCombinations(DestinationPim $pim): array
+    {
+        $query =
+<<<SQL
+SELECT f.code as family_code, f.id as family_id,
+(
+	SELECT GROUP_CONCAT(DISTINCT a.code ORDER BY a.code ASC SEPARATOR ',')
```

Comment:
a matter of style I suppose, but still weird to do request in a selector

Reply:
It's mostly effective. Not to be done in another context as a web site, but Transporteo is a "one shot" command, and it makes sense to do pragmatic choices.